sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_appinsights_utils.py

```diff
@@ -0,0 +1,535 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import logging
+import random
+import uuid
+from typing import Tuple
+
+from azure.ai.ml._azure_environments import _get_base_url_from_metadata
+from azure.ai.ml._vendor.azure_resources._resource_management_client import ResourceManagementClient
+from azure.ai.ml.constants._common import ArmConstants
+from azure.core.credentials import TokenCredential
+
+module_logger = logging.getLogger(__name__)
+
+
+def get_default_resource_group_for_app_insights(credentials: TokenCredential, subscription_id: str, location: str):
+    client = ResourceManagementClient(
+        credential=credentials,
+        subscription_id=subscription_id,
+        base_url=_get_base_url_from_metadata(),
+        api_version=ArmConstants.AZURE_MGMT_RESOURCE_API_VERSION,
+    )
+    default_rgName = get_default_resource_group_name(location)
+    # rgs = client.resource_groups.get("""try to get the default resource group""")
+    try:
+        rgs = client.resource_groups.get(resource_group_name=default_rgName)
+        return rgs.name
+    except:
+        created_rg = client.resource_groups.create_or_update(default_rgName, {"location":location})
```

Comment:
Shouldn't RG creation be part of the ARM template?

Reply:
I am creating it here since this is to create the default resource group that is needed to create the log analytics resource. It would need to exist before creating the log analytics resource inside of that resource group with the ARM template.

Reply:
Updated to include resource group creation in ARM template as a deployment, added default resource group deployment to depends on for log analytics resource